nucliadb/src/nucliadb/search/search/pgcatalog.py

```
@@ -0,0 +1,211 @@
+    else:
+        raise ValueError(f"Invalid operator {op}")
+
+
+def _prepare_query(query_parser: QueryParser):
+    filter_sql = ["kbid = %(kbid)s"]
+    filter_params: dict[str, Any] = {"kbid": query_parser.kbid}
+
+    if query_parser.query:
+        # This is doing tokenization inside the SQL server (to keep the index updated). We could move it to
+        # the python code at update/query time if it ever becomes a problem but for now, a single regex
+        # executed per query is not a problem.
+        filter_sql.append(
+            "regexp_split_to_array(lower(title), '\\W') @> regexp_split_to_array(lower(%(query)s), '\\W')"
+        )
+        filter_params["query"] = query_parser.query
+
+    if query_parser.range_creation_start:
+        filter_sql.append("created_at > %(created_at_start)s")
+        filter_params["created_at_start"] = query_parser.range_creation_start
+
+    if query_parser.range_creation_end:
+        filter_sql.append("created_at < %(created_at_end)s")
+        filter_params["created_at_end"] = query_parser.range_creation_end
+
+    if query_parser.range_modification_start:
+        filter_sql.append("modified_at > %(modified_at_start)s")
+        filter_params["modified_at_start"] = query_parser.range_modification_start
+
+    if query_parser.range_modification_end:
+        filter_sql.append("modified_at < %(modified_at_end)s")
+        filter_params["modified_at_end"] = query_parser.range_modification_end
+
+    if query_parser.filters:
+        filter_sql.append(_convert_filter(query_parser.filters, filter_params))
+
+    if query_parser.sort:
+        if query_parser.sort.field == SortField.CREATED:
+            order_field = "created_at"
+        elif query_parser.sort.field == SortField.MODIFIED:
+            order_field = "modified_at"
+        elif query_parser.sort.field == SortField.TITLE:
+            order_field = "title"
+        else:
+            raise ValueError("Unsupported sort")
```

deprecate sort by score and try not to fail hard and pick another sort field instead